File: L04_CleanCode/TypeScript/Program.ts
namespace L04_CleanCode
{
    class Class1
    {
        public static Calculate(amount: number, type: number, years: number): number
        {
            let result: number = 0;
            let disc: number = (years > 5) ? 5/100 : years/100; 
            if (type == 1)
            {
                result = amount;
            }
            else if (type == 2)
            {
                result = (amount - (0.1 * amount)) - disc * (amount - (0.1 * amount));
            }
            else if (type == 3)
            {
                result = (0.7 * amount) - disc * (0.7 * amount);
            }
            else if (type == 4)
            {
                result = (amount - (0.5 * amount)) - disc * (amount - (0.5 * amount));
            }
            return result;
        }
    }    

    console.log("Result: " + Class1.Calculate(125, 2, 5));
}

File: L04_CleanCode/TypeScript/Program.cs
using System;

namespace L04_CleanCode
{

    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Result: " + Class1.Calculate(125, 2, 5));
        }
    }


    public class Class1
    {
        public static decimal Calculate(decimal amount, int type, int years)
        {
            decimal result = 0;
            decimal disc = (years > 5) ? (decimal)5/100 : (decimal)years/100; 
            if (type == 1)
            {
                result = amount;
            }
            else if (type == 2)
            {
                result = (amount - (0.1m * amount)) - disc * (amount - (0.1m * amount));
            }
            else if (type == 3)
            {
                result = (0.7m * amount) - disc * (0.7m * amount);
            }
            else if (type == 4)
            {
                result = (amount - (0.5m * amount)) - disc * (amount - (0.5m * amount));
            }
            return result;
        }
    }    
}
